project-service/api.md

```diff
@@ -0,0 +1,594 @@
+# Project Service External API
+
+## Users
+
+### Fetch Users
+
+Returns json data of all users as an array
+
+- URL
+  + `/users`
+- Method
+  + `GET`
+- URL Params
+  + None
+- Data Params
+  + None
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+[
+  {
+    id: 0,
+    auth0Id: "google-oauth2|103547991597142817347",
+    username: "wesley"
+  }, {
+    id: 1,
+    auth0Id: "office365|10030000838D23AF@MicrosoftOnline.com",
+    username: "turtle"
+  }
+]
+```
+
+- Error Response
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Create User
+
+Create a user using information from decoded JWT
+
+- URL
+  + `/users`
+- Method
+  + `POST`
+- URL Params
+  + None
+- Data Params
+  + None
+- Success Response
+  + Code: `201 CREATED`
+  + Content:
+
+```
+{
+  id: 0,
+  auth0Id: "google-oauth2|103547991597142817347",
+  username: "wesley"
+}
+```
+
+- Error Response
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+## Projects
+
+### Create Project
+
+Create a new project. The project will by default be added to the user (as identified through JWT).
+
+- URL
+  + `/projects`
+- Method
+  + `POST`
+- URL Params
+  + None
+- Data Params
+  + Required
+    * `name=[string]`
+- Success Response
+  + Code: `201 CREATED`
+  + Content:
+
+```
+{
+  id: 1,
+  name: 'thesis'
+  createdAt: [date]
+}
+```
+
+- Error Response
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Fetch a User's Projects
+
+Get all existing projects for a user (as identified through JWT).
+
+- URL
+  + `/projects`
+- Method
+  + `GET`
+- URL Params
+  + None
+- Data Params
+  + None
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+[
+  {
+    id: 0,
+    name: 'greenfield'
+  }, {
+    id: 1,
+    name: 'legacy'
+  }
+]
+```
+
+- Error Response
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Fetch a Project's Users, Sprints and Tasks
+
+If `projectId` is specified, this end point will return all users, sprints and tasks of a project.
+
+- URL
+  + `/projects`
```

Comment:
when we are writing urls, maybe write the params in the url?

`/projects/:project_id`?

(example: `/projects/15`)?